### ISSUE TITLE
Broken import for datastore-storage-adapter on fresh Expo RN app

### ISSUE THREAD

**Issue by myfoostrong** [aws-amplify/amplify-js#12801]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Authentication, REST API, DataStore, Storage

### Amplify Version

v6

### Amplify Categories

auth, storage, api, Not applicable

### Backend

Amplify CLI

### Environment information

<details>

```
# Put output below this line
System:
    OS: macOS 13.4.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 38.96 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.0 - ~/.nvm/versions/node/v18.19.0/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v18.19.0/bin/yarn
    npm: 10.2.3 - ~/.nvm/versions/node/v18.19.0/bin/npm
    Watchman: 2023.12.04.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 120.0.6099.199
    Safari: 16.5.2
  npmPackages:
    @aws-amplify/datastore-storage-adapter: ^2.1.10 => 2.1.10 
    @aws-amplify/react-native: ^1.0.10 => 1.0.10 
    @babel/core: ^7.20.0 => 7.23.7 
    @react-native-async-storage/async-storage: 1.18.2 => 1.18.2 
    @react-native-community/netinfo: 9.3.10 => 9.3.10 
    HelloWorld:  0.0.1 
    aws-amplify: ^6.0.10 => 6.0.10 
    aws-amplify/adapter-core:  undefined ()
    aws-amplify/analytics:  undefined ()
    aws-amplify/analytics/kinesis:  undefined ()
    aws-amplify/analytics/kinesis-firehose:  undefined ()
    aws-amplify/analytics/personalize:  undefined ()
    aws-amplify/analytics/pinpoint:  undefined ()
    aws-amplify/api:  undefined ()
    aws-amplify/api/server:  undefined ()
    aws-amplify/auth:  undefined ()
    aws-amplify/auth/cognito:  undefined ()
    aws-amplify/auth/cognito/server:  undefined ()
    aws-amplify/auth/enable-oauth-listener:  undefined ()
    aws-amplify/auth/server:  undefined ()
    aws-amplify/datastore:  undefined ()
    aws-amplify/in-app-messaging:  undefined ()
    aws-amplify/in-app-messaging/pinpoint:  undefined ()
    aws-amplify/push-notifications:  undefined ()
    aws-amplify/push-notifications/pinpoint:  undefined ()
    aws-amplify/storage:  undefined ()
    aws-amplify/storage/s3:  undefined ()
    aws-amplify/storage/s3/server:  undefined ()
    aws-amplify/storage/server:  undefined ()
    aws-amplify/utils:  undefined ()
    expo: ~49.0.15 => 49.0.21 
    expo-status-bar: ~1.6.0 => 1.6.0 
    react: 18.2.0 => 18.2.0 
    react-native: 0.72.6 => 0.72.6 
    react-native-get-random-values: ~1.9.0 => 1.9.0 
    react-native-sqlite-storage: ^6.0.1 => 6.0.1 
  npmGlobalPackages:
    corepack: 0.22.0
    ignite-cli: 9.2.0
    npm: 10.2.3
    yarn: 1.22.21

```

</details>

### Describe the bug

Using a newly generated application, following [the Datastore web guide](https://docs.amplify.aws/react-native/build-a-backend/more-features/datastore/set-up-datastore/#setup-local-development-environment) the app fails to load the bundle with a missing reference to the library `aws-amplify-datastore-sqlite-adapter-expo`

Looking in the directory mentioned, `./node_modules/@aws-amplify/datastore-storage-adapter/dist/` I see the the expected file, `aws-amplify-datastore-sqlite-adapter-expo[.min].js`. Changing the 

### Expected behavior

The application to successfully load in Expo Go.

### Reproduction steps

1. Run commands as per guide, and load app in emulator
```
npx create-expo-app AmplifyDataStoreExpo
cd AmplifyDataStoreExpo
expo install aws-amplify @aws-amplify/react-native @react-native-community/netinfo @aws-amplify/datastore-storage-adapter react-native-sqlite-storage @react-native-async-storage/async-storage react-native-get-random-values
yarn ios
````
2. Add code to `App.js`
```
import { DataStore } from 'aws-amplify/datastore';
import { ExpoSQLiteAdapter } from '@aws-amplify/datastore-storage-adapter/ExpoSQLiteAdapter';

DataStore.configure({
  storageAdapter: ExpoSQLiteAdapter
});
```
3. Receive error seen in log output

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line
 ERROR  [Error: undefined Unable to resolve module ../dist/aws-amplify-datastore-sqlite-adapter-expo.min.js from /XXX/AmplifyDataStoreExpo/node_modules/@aws-amplify/datastore-storage-adapter/ExpoSQLiteAdapter/index.js: 

None of these files exist:
  * node_modules/@aws-amplify/datastore-storage-adapter/dist/aws-amplify-datastore-sqlite-adapter-expo.min.js(.native|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs)
  * node_modules/@aws-amplify/datastore-storage-adapter/dist/aws-amplify-datastore-sqlite-adapter-expo.min.js/index(.native|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs)
  4 |
  5 | if (process.env.NODE_ENV === 'production') {
> 6 | 	module.exports = require('../dist/aws-amplify-datastore-sqlite-adapter-expo.min.js');
    | 	                          ^
  7 | } else {
  8 | 	module.exports = require('../dist/aws-amplify-datastore-sqlite-adapter-expo.js');
  9 | }]


```

</details>

### Mobile Device

iPhone14 Emulator

### Mobile Operating System

iOS16.4

### Mobile Browser

Expo Go

**Comment by danrivett** [comment]
I can confirm we've also run into this exact same error, and so have temporarily removed using that adapter in our internal testing of our v5 -> v6 migration.

Would love to switch back to the Sqlite adapter soon as I believe query performance in particular is much better in an RN app.

**Comment by Oyetayo** [comment]
Experiencing the same issue. After several version tests, only version 2.0.30 seems to be stable. All others are in this Path: datastore-storage-adapter/dist/umd  not datastore-storage-adapter/dist/

**Comment by myfoostrong** [comment]
> Experiencing the same issue. After several version tests, only version 2.0.30 seems to work . All others are in this Path: datastore-storage-adapter/dist/umd not datastore-storage-adapter/dist/ and when you manually change it, it fails when you they to build the app.

This worked for me as well, both in a fresh Expo app as well as my exisiting app

**Comment by Oyetayo** [comment]
> > Experiencing the same issue. After several version tests, only version 2.0.30 seems to work. All others are in this Path: datastore-storage-adapter/dist/umd not datastore-storage-adapter/dist/ and when you manually change it, it fails when you they to build the app.
> 
> This worked for me as well, both in a fresh Expo app as well as my existing app

Yeah, but I don't think it is efficient, as my app still loads from the cloud directly and takes time to process queries. This looks more like an indication that the offline functionality of datastore is not working. I've been sweating my ass for days now looking for a workaround.

**Comment by Oyetayo** [comment]
@david-mcafee pls can you help me out with this:
I've been encountering slow query performance in my app's AWS Amplify DataStore since it started getting more usage. Despite my efforts to troubleshoot, I haven't been able to find a solution. I suspect that the performance degradation might be linked to the aws-amplify-datastore-sqlite-adapter-expo. Here are the dependencies I'm currently using:

 "@aws-amplify/datastore-storage-adapter": "^2.0.30",
 "@react-native-async-storage/async-storage": "1.17.11",
 "aws-amplify": "^5.3.14",
 "expo": "~48.0.15",

**Comment by chrisbonifacio** [comment]
Hi @myfoostrong 👋  thanks for raising this issue. I think I've run into this recently as well. Have you tried running `npx expo run:ios` instead?

**Comment by Oyetayo** [comment]
Hi @chrisbonifacio, the problem I see with this is that even if it is resolved with npx expo run: ios to run on the simulator, when you run eas build one would encounter the same issue there.

**Comment by danrivett** [comment]
In addition to fixing this broken import, Expo has just released SDK 50 which introduces a big update to their `sqlite` capability (see their release notes [here](https://expo.dev/changelog#new-and-improved-sqlite-and-camera-apis)).

I don't want to muddy the waters with this ticket, but other than say I hope DataStore supports sqlite well on Expo platforms for a long time and that the above Expo change helps this over time.

Looking forward to updates to this ticket with a fix.

**Comment by Makatun** [comment]
Same issue

**Comment by jgo80** [comment]
Same issue using Expo 50

**Comment by Makatun** [comment]
This helped me:
npm install expo@^50.0.3
npx expo install expo-sqlite

**Comment by raihanNoman** [comment]
tl;dr : It's a file path issue. Idk how to contact aws team. help!

Follow this file path: 
``` node-modules > @aws-amplify > dtatastore-storage-adapter > ExpoSQLiteAdapter > index.js```

The file contains: 
```js
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
'use strict';

if (process.env.NODE_ENV === 'production') {
	module.exports = require('../dist/aws-amplify-datastore-sqlite-adapter-expo.min.js');
} else {
	module.exports = require('../dist/aws-amplify-datastore-sqlite-adapter-expo.js');
}

```



Currently, the folder arrangement looks like this: 
```"@aws-amplify/datastore-storage-adapter": "^2.1.13"``` -Latest version

<img width="426" alt="Screenshot 2024-01-31 at 10 06 13 PM" src="https://github.com/aws-amplify/amplify-js/assets/92328461/ef73f569-0fd7-413b-b5b7-45ec8d0b92ad">




In one of my older projects 3 months ago , the folder arrangement looked like this: 
```"@aws-amplify/datastore-storage-adapter": "^2.0.40"```

<img width="526" alt="Screenshot 2024-01-31 at 10 04 47 PM" src="https://github.com/aws-amplify/amplify-js/assets/92328461/96619a9e-96e8-4380-9dd9-85f532098f52">

Help me contact AWS Team to change the modile.exports to the correct file path.

**Comment by raihanNoman** [comment]
Temporary Fix: 

```npm install @aws-amplify/datastore-storage-adapter@2.0.40```

**Comment by jgo80** [comment]
I think it's obvious and super simple to reproduce this error, what can we do to speed up this issue and tag it as a bug?

**Comment by chrisbonifacio** [comment]
Apologies for the delayed. We're able to reproduce this internally.

Here's a minimum repro app for the team:
https://github.com/chrisbonifacio/amplify-expo-app

**Comment by 0nei** [comment]
> Apologies for the delay. We're able to reproduce this internally.
> 
> Here's a minimum repro app for the team: https://github.com/chrisbonifacio/amplify-expo-app
> 
> In the meantime, please use `@aws-amplify/datastore-storage-adapter@2.0.40`. Not sure if necessary, might just be something I did wrong, but I also had to install `expo-sqlite` alongside it to work.
> 
> I kept getting `requireNativeModule` ExpoSQLite errors otherwise.

I also had to make sure `expo-sqlite` was installed for it to work, but otherwise using DataStore in my app appears to be working just fine with this temporary fix.

**Comment by amitchaudhary140** [comment]
> @aws-amplify/datastore-storage-adapter@2.0.40

@chrisbonifacio When do you (team) plan to fix the issue related to the path? Also, switching to 2.0.40 is worth in terms of performance?

**Comment by david-mcafee** [comment]
I currently have a PR open to fix the broken subpaths, but I also wanted to note that Expo Go is currently not officially supported by Amplify V6. However, Expo and React Native CLI apps are supported.

**Comment by pascalito007** [comment]
any update ? keep having the same issue

**Comment by ChadyG** [comment]
Getting the same issue today.  Using react native CLI (not expo).  Excerpt from package-lock below

`
"dependencies": {
        "@aws-amplify/datastore-storage-adapter": "^2.1.84",
        "@aws-amplify/react-native": "^1.1.10",
        "@azure/core-asynciterator-polyfill": "^1.0.2",
        "@react-native-async-storage/async-storage": "^2.2.0",
        "@react-native-community/netinfo": "^11.4.1",
        "aws-amplify": "^6.15.1",
        "react": "19.0.0",
        "react-native": "0.79.2",
        "react-native-get-random-values": "^1.11.0",
        "react-native-sqlite-storage": "^6.0.1"
      }
`

Build log errors:

`UnableToResolveError: Unable to resolve module ../dist/aws-amplify-datastore-storage-adapter.min.js from /Users/chadyg/Projects/FullSwing/ios_rn/RNTest/node_modules/@aws-amplify/datastore-storage-adapter/SQLiteAdapter/index.js: 

None of these files exist:
  * node_modules/@aws-amplify/datastore-storage-adapter/dist/aws-amplify-datastore-storage-adapter.min.js(.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx)
  * node_modules/@aws-amplify/datastore-storage-adapter/dist/aws-amplify-datastore-storage-adapter.min.js`

**Comment by richstimson** [comment]
Environment: Expo/React Native, Android device. 
    "expo": "~51.0.28",
    "@aws-amplify/datastore-storage-adapter": "^2.0.30",
    "aws-amplify": "^6.15.1",
    "react-native": "0.74.5"
    "@azure/core-asynciterator-polyfill": "^1.0.2",
    "@react-native-async-storage/async-storage": "1.23.1",


I copied the files from datastore-storage-adapter/dist/umd to datastore-storage-adapter/dist. This fixes the build error. 
Downgraded to @aws-amplify/datastore-storage-adapter@2.0.30 (and tried 2.0.40).
Follwed the instructions here: https://docs.amplify.aws/gen1/react-native/build-a-backend/more-features/datastore/set-up-datastore.

The app runs ok, but still stores data is async storage - 

[2025-06-20T23:02:06.027Z] DBG: DOCUMENT DIR (INC SQLITE) ------------------------------------
[2025-06-20T23:02:06.069Z] File: rList, size: 97 bytes
[2025-06-20T23:02:06.167Z] File: local-storage.json, size: 97 bytes
[2025-06-20T23:02:06.208Z] File: phenotype/shared/com.google.android.gms.maps#com.anonymous.locapp23.pb does not exist
[2025-06-20T23:02:06.214Z] File: DevLauncherApp-BridgeReactNativeDevBundle.js, size: 16350366 bytes
[2025-06-20T23:02:06.220Z] File: com.google.android.gms.maps._m_u, size: 0 bytes
[2025-06-20T23:02:06.224Z] File: _m_t, size: 234 bytes
[2025-06-20T23:02:06.229Z] File: DATA_ServerControlledParametersManager.data.v1.com.anonymous.locapp23, size: 30482 bytes
[2025-06-20T23:02:06.253Z] File: phenotype_storage_info/shared/storage-info.pb, size: 98 bytes
[2025-06-20T23:02:06.259Z] File: SavedClientParameters.data.cs, size: 40306 bytes

[2025-06-20T23:02:06.260Z] DBG: ASYNC STORAGE ------------------------------------
[2025-06-20T23:02:06.490Z] All AsyncStorage keys: Array(7060)
[2025-06-20T23:02:12.045Z] Key: @AmplifyDatastore::datastore_Setting::Data::01JY7S6HD5VJ5AD6YBCETZJCP6::01JY7S65TN715422K3HFC7GTTA
Value: {"key":"schemaVersion","value":"\"39950fd8ae5daccedaf94af4e406130d\"","id":"01JY7S65TN715422K3HFC7GTTA"}
[2025-06-20T23:02:12.238Z] Key: @AmplifyDatastore::sync_ModelMetadata::Data::01JY7S6HFRDGF3AZC9XCNTPW6A::01JY7S65TN715422K3HFC7GTTB
Value: {"model":"Todo","namespace":"user","lastSync":null,"fullSyncInterval":86400000,"lastFullSync":null,"lastSyncPredicate":null,"id":"01JY7S65TN715422K3HFC7GTTB"}

[2025-06-20T23:02:12.465Z] Key: @AmplifyDatastore::sync_ModelMetadata::Data::01JY7S6HGJW5SNJJGNW2VJDB84::01JY7S65TN715422K3HFC7GTTC
Value: {"model":"User","namespace":"user","lastSync":null,"fullSyncInterval":86400000,"lastFullSync":null,"lastSyncPredicate":null,"id":"01JY7S65TN715422K3HFC7GTTC"}

[2025-06-20T23:02:17.297Z] Key: @AmplifyDatastore::sync_ModelMetadata::Data::01JY7S6HH5MY2ZA9JNHDG9CMJ7::01JY7S65TN715422K3HFC7GTTD
Value: {"model":"GeoEndpointsMap","namespace":"user","lastSync":null,"fullSyncInterval":86400000,"lastFullSync":null,"lastSyncPredicate":null,"id":"01JY7S65TN715422K3HFC7GTTD"}

[2025-06-20T23:02:17.455Z] Key: @AmplifyDatastore::sync_ModelMetadata::Data::01JY7S6HHKRNTF72JECPWD398S::01JY7S65TN715422K3HFC7GTTE
Value: {"model":"Vendor","namespace":"user","lastSync":null,"fullSyncInterval":86400000,"lastFullSync":null,"lastSyncPredicate":null,"id":"01JY7S65TN715422K3HFC7GTTE"} 

.... etc ....

I also get SQLITE_FULL, error code 13. The device has lots of free storage. 

Is this related to the issue here? Any ideas on what the problem could be?
